### ISSUE TITLE
Browse zip file entries without loading data into memory

### ISSUE THREAD

**Issue by manderson23** [Stuk/jszip#290]
From reading the API docs it doesn't seem possible to iterate though entry meta data (filename etc.) without reading the complete zip file into browser memory.

It would be useful to browse the zip file metadata without loading the complete file. Like getEntries in zip.js (http://gildas-lormeau.github.io/zip.js/core-api.html#zip-reading).

**Comment by dduponchel** [comment]
The missing feature seems to be the custom (async) reader. In the given link, only `HttpRangeReader` seems to not need everything in memory. Or do you see an other solution to avoid that in a browser ?

**Comment by manderson23** [comment]
Maybe just giving that link wasn't quite clear enough.

When testing zip.js I was able to do the something like:

    zip.createReader(new zip.BlobReader(blob), function(reader) {

      // get all entries from the zip
      reader.getEntries(function(entries) {
        if (entries.length) {
          console.log(entries[0]);
        }
      });
    });

to list the entries in the zip file. I was able to do this for a 3Gb zip file with browser memory usage staying pretty much flat.

From reading the jszip API documents I couldn't see an equivalent to this.

**Comment by dduponchel** [comment]
Indeed, we read the full content of the blob before parsing it. This requires some changes in the way we read blob objects but this would definitively be a nice thing to do.

**Comment by vincerubinetti** [comment]
I'm looking to do this. I have very large (like gigabytes) zip files that I'd like to get a file listing of (filenames, extensions, sizes if possible, etc.) without downloading them. I found this Python library that does it using Http ranges:

[unzip-http](https://github.com/saulpw/unzip-http)

I think I'm going to try to port their `infoiter` function to JavaScript. I'll post here if I can come up with a snippet that works.

Unless anyone knows of any existing JS library that can already do this? I searched npm pretty thoroughly and couldn't find anything.
